Upstream: Yes
Source: https://chromium-review.googlesource.com/c/chromium/src/+/2153148

From ae057473baa2ffec0fac7eed529a13047b2f50b0 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Fri, 17 Apr 2020 06:41:44 +0000
Subject: [PATCH] libstdc++: make NGPhysicalContainerFragment::ConstIterator usable with std::all_of

Use of std::all_of requires STL compliant iterator. However,
NGPhysicalContainerFragment::ConstIterator does not define
iterator_tag and therefore is no STL iterator. To make it
compliant derive NGPhysicalContainerFragment::ConstIterator from
std::iterator.

Bug: 957519
Change-Id: Id042f987ca093ece6cd1cd10f601eb1909700d08
---

diff --git a/third_party/blink/renderer/core/layout/ng/ng_physical_container_fragment.h b/third_party/blink/renderer/core/layout/ng/ng_physical_container_fragment.h
index b3c7624..1256e77 100644
--- a/third_party/blink/renderer/core/layout/ng/ng_physical_container_fragment.h
+++ b/third_party/blink/renderer/core/layout/ng/ng_physical_container_fragment.h
@@ -5,6 +5,8 @@
 #ifndef THIRD_PARTY_BLINK_RENDERER_CORE_LAYOUT_NG_NG_PHYSICAL_CONTAINER_FRAGMENT_H_
 #define THIRD_PARTY_BLINK_RENDERER_CORE_LAYOUT_NG_NG_PHYSICAL_CONTAINER_FRAGMENT_H_
 
+#include <iterator>
+
 #include "base/containers/span.h"
 #include "third_party/blink/renderer/core/core_export.h"
 #include "third_party/blink/renderer/core/layout/geometry/physical_rect.h"
@@ -31,7 +33,8 @@
     PostLayoutChildLinkList(wtf_size_t count, const NGLink* buffer)
         : count_(count), buffer_(buffer) {}
 
-    class ConstIterator {
+    class ConstIterator
+        : public std::iterator<std::input_iterator_tag, NGLink> {
       STACK_ALLOCATED();
 
      public:
